bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/discovery/AmazonEchoDiscovery.java

```diff
@@ -188,30 +170,13 @@ public synchronized void discoverFlashBriefingProfiles(String currentFlashBriefi
         if (currentFlashBriefingJson.isEmpty()) {
             return;
         }
-        DiscoveryServiceCallback discoveryServiceCallback = this.discoveryServiceCallback;
-        if (discoveryServiceCallback == null) {
-            return;
-        }
 
         if (!discoverdFlashBriefings.contains(currentFlashBriefingJson)) {
-            ThingUID freeThingUID = null;
-            int freeIndex = 0;
-            for (int i = 1; i < 1000; i++) {
-                String id = Integer.toString(i);
-                ThingUID brigdeThingUID = this.accountHandler.getThing().getUID();
-                ThingUID thingUID = new ThingUID(THING_TYPE_FLASH_BRIEFING_PROFILE, brigdeThingUID, id);
-                if (discoveryServiceCallback.getExistingThing(thingUID) == null
-                        && discoveryServiceCallback.getExistingDiscoveryResult(thingUID) == null) {
-                    freeThingUID = thingUID;
-                    freeIndex = i;
-                    break;
-                }
-            }
-            if (freeThingUID == null) {
-                logger.debug("No more free flashbriefing thing ID found");
-                return;
-            }
-            DiscoveryResult result = DiscoveryResultBuilder.create(freeThingUID).withLabel("FlashBriefing " + freeIndex)
+            ThingUID brigdeThingUID = this.accountHandler.getThing().getUID();
+            ThingUID thingUID = new ThingUID(THING_TYPE_FLASH_BRIEFING_PROFILE, brigdeThingUID,
+                    Integer.toString(currentFlashBriefingJson.hashCode()));
```

Comment:
This should fix discovery of multiple flash briefings. Because of the deprecation I think it would currently always create only one briefing with id 0. :smile: